controllers/sms-gateway.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * This module implements GET and POST to support medic-gateway's API
+ * @see https://github.com/medic/medic-gateway
+ */
+
+var _ = require('underscore'),
+    db = require('../db'),
+    http = require('http'),
+    querystring = require('querystring'),
+    utils = require('./utils'),
+    messageUtils = require('./messages');
+
+require('lie/polyfill');
+
+function warn() {
+  var args = Array.prototype.slice.call(arguments, 0);
+  args.unshift('WARN', 'sms-gateway');
+  console.error.apply(console, args);
+}
+
+function readBody(stream) {
+  var body = '';
+  return new Promise(function(resolve, reject) {
+    stream.on('data', function(data) {
+      body += data.toString();
+    });
+    stream.on('end', function() {
+      resolve(body);
+    });
+    stream.on('error', reject);
+  });
+}
+
+function readResBody(res) {
+  return readBody(res)
+    .then(function(body) {
+      if (res.statusCode >= 400) {
+        throw new Error('Bad status received: ' + res.statusCode + ': ' + body);
+      }
+      return body;
+    });
+}
+
+function saveToDb(gatewayRequest, wtMessage) {
+  var messageBody = querystring.stringify({
+    from: wtMessage.from,
+    message: wtMessage.content,
+    'medic-gateway_id': wtMessage.id,
+  });
+
+  return new Promise(function(resolve, reject) {
+    var req = http.request(
+      {
+        hostname: 'localhost',
+        port: 5984,
+        path: '/medic/_design/medic/_rewrite/add',
```

Comment:
Of course not, see previous comment. Either POST /api/v1/records or require('./records').createRecord() etc...
"I'd prefer if you used /api/v1/records if possible. Either via http or require the controller... we can open an issue for refactoring common controller bits."